chia/server/server.py

```diff
@@ -263,13 +263,17 @@ async def start_server(self, on_connect: Callable = None):
                 self.chia_ca_crt_path, self.chia_ca_key_path, self.p2p_crt_path, self.p2p_key_path, log=self.log
             )
 
+        # If self._port is set to zero, the socket will bind to a new available port. Therefore, we have to obtain
+        # this port from the socket itself and update self._port.
         self.site = web.TCPSite(
             self.runner,
-            port=self._port,
+            host="0.0.0.0",
+            port=int(self._port),
             shutdown_timeout=3,
             ssl_context=ssl_context,
         )
         await self.site.start()
+        self._port = self.runner.addresses[0][1]
```

Comment:
what's at `addresses[1]` and at `addresses[0][0]`?

Reply:
addresses that are listening for this running.
[0][0] is host, [0][1] is port